### PR TITLE
[stable10] Move excluded files to config array

### PR DESCRIPTION
**Pull request by sharidas**
Moving excluded files to config array.
Instead of populating in the ExcludeFile*
php files, we can have it in config.php.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

This change would help users to mention folders/files to be excluded during integrity check to the config.php. In the config.php it will be represented as an array.

## Related Issue

https://github.com/owncloud/core/issues/23186
https://github.com/owncloud/enterprise/issues/1874

## Motivation and Context

Instead of hard coding the files/folders in the files, its better to move them to the config.php. Also this change would help the integrity check to exclude files/folders mentioned in the config.php

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1: if you updated this with the latest changes from the master PR (PHPDoc changes I believe)

**Comment by phil-davis** [comment]
This is a backport of #29126

**Comment by sharidas** [comment]
@PVince81 Yes I have updated this PR from the master branch. Replaced `mixed` with `array` in the PHPDoc - https://github.com/owncloud/core/pull/29460/files#diff-8c88c803f1798e69a75cd73cbaba2a12R233

**Comment by bcutter** [comment]
Hell I´m searching for the syntax in config.php for 20 minutes because the [mentioned](https://doc.owncloud.org/server/10.0/admin_manual/release_notes.html) config.sample.php does NOT contain any information regarding this... pretty frustrating. Is it my fault or is it just poor documentation?

**Comment by PVince81** [comment]
@bcutter it's called "integrity.excluded.files", you can see it directly in config.sample.php.

I see that the documentation hasn't been synced with the contents of config.sample.php, raised https://github.com/owncloud/documentation/issues/3638.

Thanks for spotting.

**Comment by bcutter** [comment]
Yes the name is known, but what‘s the syntax vor keys? And values absolute or relative paths? Maybe you can just link the config.sample.php here.

**Comment by PVince81** [comment]
https://github.com/owncloud/core/blob/v10.0.4/config/config.sample.php#L1246

It's relative paths

File: lib/private/IntegrityCheck/Checker.php [hunk 1]
```diff
@@ -382,13 +382,20 @@ private function verify($signaturePath, $basePath, $certificateCN, $force = fals
 			throw new InvalidSignatureException('Signature could not get verified.');
 		}
 
+		//Exclude files which shouldn't fall for comparison
```
Comment (settermjd) [inline lib/private/IntegrityCheck/Checker.php]: In this line, it seems "fall" is incorrect or unclear. @sharidas, can you clarify, please?

**Comment by bcutter** [comment]
Hmm, why it´s not working?

**config.php:**
```
  'integrity.excluded.files' =>
  array (
    '.htaccess_OLD',
        ),
```

**Integrity check after rescan is still:**
```
	- EXTRA_FILE
		- .htaccess_OLD
```

**Pull request merged**